### PR TITLE
Add computed fields and order based on is_archived

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

https://github.com/AlexsLemonade/resources-portal/issues/315

## Purpose/Implementation Notes

I'm not sure why everything is broken. I think it all went wrong when I tried to add is_archived to elasticsearch.

Other than that I added is_active, action_required_sharer, and action_required_requester to the model/view for material request.

I also change PENDING -> OPEN.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

The unit tests cover this well enough I think!

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/resources_portal/models/material_request.py [hunk 1]
```diff
@@ -64,14 +64,38 @@ class Meta:
...
+    def is_active(self):
+        return self.status in ["OPEN", "APPROVED", "IN_FULFILLMENT"]
+
+    @property
+    def requires_action_sharer(self):
+        if not self.is_active:
+            return False
+
+        if self.status == "APPROVED":
+            return not self.requires_action_requester()
```
Comment (davidsmejia) [inline api/resources_portal/models/material_request.py:83]: does this need to be executed as a function?

File: api/resources_portal/views/document_views.py [hunk 2]
```diff
@@ -253,6 +262,7 @@ class MaterialDocumentView(DocumentViewSet):
 
     # Define ordering fields
     ordering_fields = {
+        "is_archived": "is_archived",
```
Comment (davidsmejia) [inline api/resources_portal/views/document_views.py:265]: did you try `-is_archived`?

Comment (kurtwheeler) [reply api/resources_portal/views/document_views.py:265]: Hmm, that might be necessary, but the current issue I'm running into is that any materials with `is_archived` are being filtered out of the result set, not misordered.

**Comment by davidsmejia** [comment]
Just has one small conflict with `fulfillment_notes`

**Pull request merged**